src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -10358,5 +10358,330 @@ public static void M([UnscopedRef] ref S s)
+
+                class C : B
+                {
+                    public override void M({{scoped2}} {{modifier}} R r) { }
+                }
+
+                ref struct R;
+                """;
+
+            var comp = CreateCompilation(source);
+
+            if (scoped1 == "scoped" && scoped2 == "" && modifier == "ref")
+            {
+                comp.VerifyDiagnostics(
+                    // (8,26): error CS8987: The 'scoped' modifier of parameter 'r' doesn't match overridden or implemented member.
+                    //     public override void M( ref R r) { }
+                    Diagnostic(ErrorCode.ERR_ScopedMismatchInParameterOfOverrideOrImplementation, "M").WithArguments("r").WithLocation(8, 26));
+            }
+            else
+            {
+                comp.VerifyDiagnostics();
+            }
+        }
+
+        [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/76100")]
+        [InlineData("[System.Diagnostics.CodeAnalysis.UnscopedRef]", "")]
+        [InlineData("[System.Diagnostics.CodeAnalysis.UnscopedRef]", "[System.Diagnostics.CodeAnalysis.UnscopedRef]")]
+        [InlineData("", "")]
+        public void SelfAssignment_ScopeVariance_UnscopedRef_Allowed_DelegateConversion(string attr1, string attr2)
+        {
+            var source = $$"""
+                D d = C.M;
+
+                delegate void D({{attr1}} ref R r);
+
+                static class C
+                {
+                    public static void M({{attr2}} ref R r) { }
+                }
+
+                ref struct R;
+                """;
+            CreateCompilation([source, UnscopedRefAttributeDefinition]).VerifyDiagnostics();
+        }
+
+        [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/roslyn/issues/76100")]
+        public void SelfAssignment_ScopeVariance_ScopedRef_DelegateConversion(
+            [CombinatorialValues("scoped", "")] string scoped1,
+            [CombinatorialValues("scoped", "")] string scoped2,
+            [CombinatorialValues("ref", "out")] string modifier)
+        {
+            var source = $$"""
+                D d = C.M;
+
+                delegate void D({{scoped1}} {{modifier}} R r);
+
+                static class C
+                {
+                    public static void M({{scoped2}} {{modifier}} R r) { }
+                }
+
+                ref struct R;
+                """;
+
+            var comp = CreateCompilation(source);
+
+            if (scoped1 == "scoped" && scoped2 == "" && modifier == "ref")
+            {
+                comp.VerifyDiagnostics(
+                    // (1,7): error CS8986: The 'scoped' modifier of parameter 'r' doesn't match target 'D'.
+                    // D d = C.M;
+                    Diagnostic(ErrorCode.ERR_ScopedMismatchInParameterOfTarget, "C.M").WithArguments("r", "D").WithLocation(1, 7));
+            }
+            else
+            {
+                comp.VerifyDiagnostics();
+            }
+        }
```

Comment:
Is there a test to verify that the following is an error? 

```csharp
interface I1 {
  ref int M();
}
ref struct S1 : I1 {
  [UnscopedRef]
  public ref int M() => throw null!;
}
```

Reply:
Yes, `RefFieldTests.UnscopedRefAttribute_InterfaceImplementation_01`.

https://github.com/dotnet/roslyn/blob/9211f22d86f192c6835f2857101229941dd83585/src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs#L25508